File: itcast_shop_realtime_etl/src/main/scala/com/itcast/shop/realtime/etl/utils/HBaseUtil.scala
package com.itcast.shop.realtime.etl.utils

import com.itcast.shop.realtime.etl.utils.pool.{ConnectionPoolConfig, HBaseConnectionPool}
import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.hbase.HBaseConfiguration


/**
  * @ description: 
  * @ author: spencer
  * @ date: 2020/12/10 11:39
  */
object HBaseUtil {

  private val config = new ConnectionPoolConfig
  config.setMaxIdle(20)
  config.setMaxIdle(5)
  config.setMaxWaitMillis(1000)
  config.setTestOnBorrow(true)

  var hbaseConfig: Configuration = HBaseConfiguration.create()
  hbaseConfig = HBaseConfiguration.create
  hbaseConfig.set("hbase.default.for.version.skip", "true")

  // 创建连接池对象
  private val pool = new HBaseConnectionPool(config, hbaseConfig)

  def getPool() = {
    pool
  }

}

File: itcast_shop_publisher/src/main/java/com/bigdata/itcast/util/HBaseUtil.java
package com.bigdata.itcast.util;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.hbase.HBaseConfiguration;
import org.apache.hadoop.hbase.client.Connection;
import org.apache.hadoop.hbase.client.ConnectionFactory;

import java.io.IOException;

/**
 * @ description:
 * @ author: spencer
 * @ date: 2020/12/24 17:49
 */
@org.springframework.context.annotation.Configuration
public class HBaseUtil {
    private Configuration config = HBaseConfiguration.create();
    Connection connection = null;
    /**
     * 获取hbase客户端连接
     * @return
     */
    public Connection getConnection(){
        try {
           connection = ConnectionFactory.createConnection(config);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return connection;
    }

    /**
     * 关闭连接
     */
    public void closeConnection(){
        if (connection.isClosed()){
            try {
                connection.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}
